templates/settings.html

```diff
@@ -6,30 +6,37 @@
 
 {% block main %}
 <div class="container">
-    <div>
-        <h4>Current Budget</h4>
-        <p>{{ daily_budget | usd }}</p>
-    </div>
     <div>
         <h4>Preferences</h4>
-        <ul>
+        <!-- <ul>
             {% for pref in current_user_prefs %}
                 <li> {{ pref }} </li>
             {% endfor %}
-        </ul>
+        </ul> -->
         <form action="/settings" method="post">
             <div>
+                {% for cuisine in all_cuisine_types %}
+                    <div class="form-check form-switch">
+                        {% if cuisine.cuisine_type in current_user_prefs %}
+                            <input class="form-check-input" type="checkbox" id="{{ cuisine.cuisine_type }}" name="cuisine_selections" value="{{ cuisine.cuisine_type }}" checked>
+                        {% else %}
+                            <input class="form-check-input" type="checkbox" id="{{ cuisine.cuisine_type }}" name="cuisine_selections" value="{{ cuisine.cuisine_type }}">
+                        {% endif %}
+                        <label class="form-check-label" for="{{ cuisine.cuisine_type }}">{{ cuisine.cuisine_type }}</label>  
+                    </div>
+                {% endfor %}
```

Comment:
Does this not work?

I checked here: <https://community.home-assistant.io/t/setting-a-variable-in-a-loop-doesnt-work/151432/2>

```suggestion
                {% for cuisine in all_cuisine_types %}
                    <div class="form-check form-switch">
                        {% if cuisine.cuisine_type in current_user_prefs %}{% set checkedVal = 'checked' %}{% endif %}
                        <input class="form-check-input" type="checkbox" id="{{ cuisine.cuisine_type }}" name="cuisine_selections" value="{{ cuisine.cuisine_type }}" {{ checkedVal }}>
                        <label class="form-check-label" for="{{ cuisine.cuisine_type }}">{{ cuisine.cuisine_type }}</label>  
                    </div>
                {% endfor %}
```

Reply:
It might even be able to be simplified to an inline check!

```python
<input class="form-check-input" type="checkbox" id="{{ cuisine.cuisine_type }}" name="cuisine_selections" value="{{ cuisine.cuisine_type }}" {{ 'checked' if cuisine.cuisine_type in current_user_prefs else '' }}>
```

Reply:
This now works. It was failing before because i forgot to set the name and value attributes correctly

Reply:
No I meant does my suggestion not work, I didn't mean your code :)

Reply:
i have applied this suggestion , thanks for the input !